### PR TITLE
Irradiance lighting effect.

### PR DESCRIPTION
**Pull request by wlbragg**
After considerable testing and tinkering, this is about the best we can do without a custom made light map and grain map. However the grain map won't work well without breaking out some of the objects into manageable pieces. For example the chassis has parts that should not have the grain applied to it like certain tight curves and the top of the dash. It doesn't project well in those places.

### CODE REVIEW

**Comment by onox** [comment]
Could you use 4-space indentation? After that do `git rebase -i HEAD~3` (assuming you're going to add a third commit) and then change the last two "pick"'s to "fixup". (Keep "pick" for "Irradiance lighting effect")

File: Models/Effects/interior/c172p-interior-light.eff [hunk 1]
```diff
@@ -0,0 +1,23 @@
+<?xml version="1.0" encoding="utf-8"?>
+
+<PropertyList>
+  <name>c172p-interior-light</name>
+  <inherits-from>Effects/model-interior</inherits-from>
+  <parameters>
+    <irradiance-map-type type="int">2</irradiance-map-type>
+    <irradiance-map-strength type="float">1</irradiance-map-strength>
+
+    <!--texture n="3">
```
Comment (onox) [inline Models/Effects/interior/c172p-interior-light.eff]: I think you need to write `<texture` instead of `<!--texture` here?

**Comment by onox** [comment]
Btw, does this fix #472?

**Comment by wlbragg** [comment]
Sorry onox, actually we aren't even using that block of code (that's why it's commented out). That is if you want a custom light map. All we are using is the two lines that start with irradiance. I don't know why I left that block of code there, just in case someone wanted to try to make a custom light map.

**Comment by wlbragg** [comment]
Yes it will close #472.

Yes, I will fix tab spaces, sorry, forgot to check it.

The block of code that is commented out is only needed if we do a custom light map.

**Comment by wlbragg** [comment]
> invalid upstream HEAD-3

**Comment by onox** [comment]
Hmm.. I see you merged `master` into `irradiance` branch. Usually you don't have to do that.

**Comment by wlbragg** [comment]
Sorry, I was unclear as to whether or not I should do that. I knew there was at least the one change to c172p.xml tail contact.

**Comment by onox** [comment]
The only reasons to merge `master` into a feature branch is to resolve conflicts before merging the branch into `master`, or if lots of commits were added to both branches and you want to test the results.

But for these small PR's, Git can merge your branch just fine.

**Comment by wlbragg** [comment]
Thanks

**Pull request merged**